internal/pkg/bulk/engine.go

```diff
@@ -153,12 +163,12 @@ func (b *Bulker) updateBulkerMap(outputName string, newBulker *Bulker) {
 // if changed, stop the existing bulker and create a new one
 func (b *Bulker) CreateAndGetBulker(ctx context.Context, zlog zerolog.Logger, outputName string, outputMap map[string]map[string]interface{}) (Bulk, bool, error) {
 	hasConfigChanged := b.hasChangedAndUpdateRemoteOutputConfig(zlog, outputName, outputMap[outputName])
-	bulker := b.bulkerMap[outputName]
-	if bulker != nil && !hasConfigChanged {
-		return bulker, false, nil
+	bulker, ok := b.bulkerMap.Load(outputName)
+	if ok && !hasConfigChanged {
+		return bulker.(Bulk), false, nil
```

Comment:
I'm assuming it is guaranteed that the sync.Map will only hold bulker type, is that the case?

Reply:
yes